### PR TITLE
Improvements for variable names

### PR DESCRIPTION
**Pull request by danielgindi**

### CODE REVIEW

File: BigEval.js [hunk 1]
```diff
@@ -298,12 +295,34 @@ BigEval.prototype._tokenizeExpression = function (expression) {
...
+
 		if (isVarChars) {
-			token += c;
+			// Starting a variable name - can start only with A-Z_
+			
+			var token = '';
+
+			while (i < len) {
+				c = expression[i];
+                isVarChars = this.varNameChars[c];
```
Comment (aviaryan) [inline BigEval.js]: I saw that our project was a mess (as seen in this line) when it came to indentation so I have fixed on following [happiness](https://github.com/JedWatson/happiness) guidelines in the project. 
Please rebase with master and then the new checks should apply on this PR too.

Comment (danielgindi) [inline BigEval.js]: Done :-)

**Comment by aviaryan** [comment]
@danielgindi Thanks for the quick work. Unfortunately, a test fails, please check.

**Comment by danielgindi** [comment]
Seems OK now.
There was one test that failed even with master, due to native `eval` not handline the `**` operator then the l-value is a negative number.
It gave "Invalid unary operator on the left hand side of exponentiation (**) operator".

I've fixed that, and fixed another issue with the exponentiation operator along the way (When there's multiple **, like `2**7**-3`), where the order should be reversed.

**Comment by danielgindi** [comment]
Btw, you may want to add a link to the port https://github.com/danielgindi/Eval.net

**Pull request merged**